src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal class LocalUsingVarRewriter : BoundTreeRewriterWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
+    {
+        public static BoundNode Rewrite(BoundStatement statement)
+        {
+            var localUsingVarRewriter = new LocalUsingVarRewriter();
+            return localUsingVarRewriter.Visit(statement);
+        }
+
+        public override BoundNode VisitBlock(BoundBlock node)
```

Comment:
Will work on this.

Reply:
Is this something we can PROTOTYPE and cleanup in a future PR? This PR is getting quite large / dated at this point. Make sense to get this merged and new PR to fix?

Reply:
I can put this in with the increased test coverage in the next PR?

Reply:
The structure that we discussed with Lauren for `VisitBlock`:
1. (optional) do a quick scan to check if there are any using-variables, bail out otherwise
2. iterate over the statements starting from the top, adding them back into a builder/stack one by one
3. but every time you encounter a single-declaration using variable, replace the rest of the statements with a `using` wrapping them, and break
4. or if you encountered a multi-declaration, produce a `using` for each one (where the block comes from wrapping the rest of the statements), and break
5. after the loop, return the statements in the builder as a `BoundBlock`

The key method would be something like `BoundBlock RewriteRest(ImmutableArray<BoundStatement> statements, int startFrom)`.